yogstation/code/modules/antagonists/flockmind/structures/cage.dm

```diff
@@ -0,0 +1,89 @@
+/obj/structure/destructible/flock/cage
+	name = "weird energy cage"
+	desc = "You can see the person inside being rapidly taken apart by fibrous mechanisms."
+	flock_desc = "Spins living matter into resources. Painfully."
+	flock_id = "matter reprocessor"
+	icon_state = "cage"
+	anchored = TRUE
+	can_buckle = TRUE
+	density = FALSE
+	can_be_unanchored = FALSE
+	canSmoothWith = null
+	flags_1 = NODECONSTRUCT_1
+	var/processing = FALSE
+	var/resources = 0
+
+/obj/structure/destructible/flock/cage/user_unbuckle_mob(mob/living/buckled_mob, mob/living/user)
+	if(has_buckled_mobs())
+		for(var/buck in buckled_mobs) //breaking a nest releases all the buckled mobs, because the nest isn't holding them down anymore
+			var/mob/living/M = buck
```

Comment:
early return + full var in the loop rather then setting it afterwards

Reply:
Done